pinot-core/src/main/java/com/linkedin/pinot/core/io/readerwriter/impl/FixedByteSingleColumnMultiValueReaderWriter.java

```diff
@@ -79,46 +107,52 @@
   private int incrementalCapacity;
   private int columnSizeInBytes;
   private int maxNumberOfMultiValuesPerRow;
+  private final int rowCountPerChunk;
+  private int prevRowStartIndex = 0;  // Offset in the databuffer for the last row added.
+  private int prevRowLength = 0;  // Number of values in the column for the last row added.
 
-  public FixedByteSingleColumnMultiValueReaderWriter(int rows, int columnSizeInBytes, int maxNumberOfMultiValuesPerRow,
-      int avgMultiValueCount)
-      throws IOException {
-    int initialCapacity = Math.max(maxNumberOfMultiValuesPerRow, rows * avgMultiValueCount);
+  public FixedByteSingleColumnMultiValueReaderWriter(int maxNumberOfMultiValuesPerRow, int avgMultiValueCount, int rowCountPerChunk,
+      int columnSizeInBytes) {
+    int initialCapacity = Math.max(maxNumberOfMultiValuesPerRow, rowCountPerChunk * avgMultiValueCount);
```

Comment:
Preconditions to assert these inputs are > 0. Definitely needed for rowCOuntPerChunk to ensure headerSize > 3 * sizeof(int)